packages/transport/src/modules/relationships/local/AuditLog.ts

```diff
@@ -0,0 +1,50 @@
+import { serialize, type, validate } from "@js-soft/ts-serval";
+import _ from "lodash";
+import { CoreAddress, CoreDate, CoreSerializable, ICoreAddress, ICoreDate } from "../../../core";
+import { AuditLogEntryReason, AuditLog as BackboneAuditLog } from "../transmission/AuditLog";
+import { RelationshipStatus } from "../transmission/RelationshipStatus";
+
+export interface IAuditLogEntry {
+    createdAt: ICoreDate;
+    createdBy: ICoreAddress;
+    reason: AuditLogEntryReason;
+    oldStatus?: RelationshipStatus;
+    newStatus: RelationshipStatus;
+}
+
+@type("AuditLogEntry")
+export class AuditLogEntry extends CoreSerializable implements IAuditLogEntry {
+    @validate()
+    @serialize()
+    public createdAt: CoreDate;
+
+    @validate()
+    @serialize()
+    public createdBy: CoreAddress;
+
+    @validate()
+    @serialize()
+    public reason: AuditLogEntryReason;
+
+    @validate({ nullable: true })
+    @serialize()
+    public oldStatus?: RelationshipStatus;
+
+    @validate()
+    @serialize()
+    public newStatus: RelationshipStatus;
+
+    public static from(value: IAuditLogEntry): AuditLogEntry {
+        return this.fromAny({ ...value, oldStatus: value.oldStatus ?? undefined });
+    }
+}
+
+export class AuditLog {
```

Comment:
I'd vote for naming this RelationshipAuditLog (and RelationshipAuditLogEntry)

Reply:
don't forget to rename the file when changing

Reply:
I feel like enmeshed is suffering from overly long names enough - my vote is for keeping it like this until a different audit log pops up

Reply:
does it need a type annotation though? I can't think of an instance where the audit log would not be inside its relationship

Reply:
@stnmtz has a strong opinion about Serializables without type annotationy, so yes - it does.